### ISSUE TITLE
Add a rule to enforce preference between `infer` and `output`

### ISSUE THREAD

**Issue by andreww2012** [marcalexiei/eslint-plugin-zod-x#182]
### 🧩 Description

Since [`infer` and `output` are actually the same type](https://github.com/colinhacks/zod/blob/9977fb0868432461de265a773319e80a90ba3e37/packages/zod/src/v4/core/core.ts#L120), it's purely a stylistic choice to choose between the two. My guess that `infer` exists for compatibility with v3, and `output` is consistent with `input`.

It would be nice to have a rule to enforce a single style.

Suggested rule name: `schema-type-style`

### ⚙️ Options

```ts
inteface Options {
  style: 'infer' | 'output';
}
```

I think the default should be `infer` as the most widely used (or maybe `output` to promote the more consistent name? I don't know 😄)

### 🧾 Example Configuration

```ts
'zod-x/schema-type-style': 2,
'zod-x/schema-type-style': [2, {style: 'infer'}],
'zod-x/schema-type-style': [2, {style: 'output'}],
```

### ✅ Valid Code Examples

```typescript
type SchemaType = z.infer<typeof Schema>
```

### ❌ Invalid Code Examples

```typescript
type SchemaType = z.output<typeof Schema>
```

**Comment by marcalexiei** [comment]
Hello again Andrew!
Now I'm working on #171.

Once completed I will add this rule to the plugin.
Probably I will name it something more verbose like `consistent-schema-output-type-style`, 
similar to `vitest/consistent-vitest-vi`.

**Comment by marcalexiei** [comment]
Available in [v3.8.0](https://github.com/marcalexiei/eslint-plugin-zod/releases/tag/v3.8.0).
